dlt/destinations/impl/sink/sink.py

```diff
@@ -0,0 +1,172 @@
+    def run(self, start_index: int) -> Iterable[TDataItems]:
+        # stream items
+        from dlt.common.libs.pyarrow import pyarrow
+
+        # guard against changed batch size after restart of loadjob
+        assert (
+            start_index % self._config.batch_size
+        ) == 0, "Batch size was changed during processing of one load package"
+
+        start_batch = start_index / self._config.batch_size
+        with pyarrow.parquet.ParquetFile(self._file_path) as reader:
+            for record_batch in reader.iter_batches(batch_size=self._config.batch_size):
+                if start_batch > 0:
+                    start_batch -= 1
+                    continue
+                batch = record_batch.to_pylist()
+                yield batch
+
+
+class SinkJsonlLoadJob(SinkLoadJob):
+    def run(self, start_index: int) -> Iterable[TDataItems]:
+        current_batch: TDataItems = []
+
+        # stream items
+        with FileStorage.open_zipsafe_ro(self._file_path) as f:
+            for line in f:
+                # find correct start position
+                if start_index > 0:
+                    start_index -= 1
+                    continue
+                current_batch.append(json.loads(line))
+                if len(current_batch) == self._config.batch_size:
+                    yield current_batch
+                    current_batch = []
+            yield current_batch
+
+
+class SinkClient(JobClientBase):
+    """Sink Client"""
+
+    capabilities: ClassVar[DestinationCapabilitiesContext] = capabilities()
+
+    def __init__(self, schema: Schema, config: SinkClientConfiguration) -> None:
+        super().__init__(schema, config)
+        self.config: SinkClientConfiguration = config
+
+    def initialize_storage(self, truncate_tables: Iterable[str] = None) -> None:
+        pass
+
+    def is_storage_initialized(self) -> bool:
+        return True
+
+    def drop_storage(self) -> None:
+        pass
+
+    def update_stored_schema(
+        self, only_tables: Iterable[str] = None, expected_update: TSchemaTables = None
+    ) -> Optional[TSchemaTables]:
+        return super().update_stored_schema(only_tables, expected_update)
+
+    def start_file_load(self, table: TTableSchema, file_path: str, load_id: str) -> LoadJob:
+        # save our state in destination name scope
+        load_state = load_package_destination_state()
+        if file_path.endswith("parquet"):
+            return SinkParquetLoadJob(table, file_path, self.config, self.schema, load_state)
+        if file_path.endswith("jsonl"):
+            return SinkJsonlLoadJob(table, file_path, self.config, self.schema, load_state)
+        return None
+
+    def restore_file_load(self, file_path: str) -> LoadJob:
+        return EmptyLoadJob.from_file_path(file_path, "completed")
+
+    def complete_load(self, load_id: str) -> None:
+        # pop all state for this load on success
+        clear_loadpackage_destination_state()
```

Comment:
IMO it is better if we do not delete it. You can use it ie. to inspect how many rows were processed.

Reply:
alright, changed!